2-building-blocks-of-oop-part-1/src/Point.ts

```diff
@@ -0,0 +1,35 @@
+export class Point {
+  private readonly x: number;
+  private readonly y: number;
+
+  private calcDistance(x: number, y: number): number {
+    return Math.sqrt(Math.pow(x - this.x, 2) + Math.pow(y - this.y, 2));
+  }
+
+  constructor();
+  constructor(x: number, y: number);
+  constructor(x = 0, y = 0) {
```

Comment:
Можно добавить модификаторы доступа прямо в конструктор, чтобы не объявлять свойства в начале класса. Это избавит также от инициализации этих свойств в конструкторе.
```suggestion
  constructor(public x: number = 0, public y: number = 0) {
```